src/app/AttributeAccessInterface.h

```diff
@@ -200,6 +200,35 @@ class AttributeValueEncoder
         return Encode(DataModel::List<uint8_t>());
     }
 
+    /**
+     * @brief Conditionaly uses the provided aCallback to encode the attribute data.
+     *        When the condition evaluates to false, it will encode null instead by using EncodeNull
+     *
+     *        aCallback is expected to take a const auto argument (which will convert to a AttributeValueEncoder pointer)
+     *        and encode, with the available method of this class, an attribute data.
+     *        Any error during its process needs to be directly returned to the user as a CHIP_ERROR
+     *
+     *        This method should only be used to encode nullable attributes
+     *        aCallback is not called when the condition evaluates to false
+     *
+     * @tparam AttributeEncoder
+     * @param condition When true, use aCallback to encode the attribute data. When false, encode null
+     * @param aCallback Provide your lamda function encoding the data.
+     * @return CHIP_ERROR Encoding succeed or failed with error code
+     */
+    template <typename AttributeEncoder>
+    CHIP_ERROR EncodeNullableOnCondition(bool condition, AttributeEncoder aCallback)
```

Comment:
The naming is pretty opaque; nothing about it says "will encode null if condition is false".

Why is this significantly better than consumers either putting their data in a Nullable and then calling `Encode` on it, or just having consumers branch on the condition themselves?

If we just want a helper for the OpenThread code, that helper could be file-local....

Reply:
I did implementation formmy use case in mind but thought it could be useful for others..maybe other diagnostic clusters attributes. Any how, based on Tennessee's comment, I compared the code size difference and it is not worth it. I'll change the code tomorrow morning .

Reply:
I removed the helper entirely